sky/utils/resource_checker.py

```diff
@@ -151,3 +136,159 @@ def get_all_managed_jobs():
+
+
+def check_users_workspaces_active_resources(
+        user_ids: List[str],
+        workspace_names: List[str]) -> Tuple[str, List[str]]:
+    """Check if all the active clusters or managed jobs in workspaces
+       belong to the user_ids. If not, return the error message.
+
+    Args:
+        user_ids: List of user_id.
+        workspace_names: List of workspace_name.
+
+    Returns:
+        resource_error_summary: str
+        missed_users_names: List[str]
+    """
+    all_clusters, all_managed_jobs = _get_active_resources_for_workspaces(
+        workspace_names)
+    resource_errors = []
+    missed_users = set()
+    active_cluster_names = []
+    active_job_names = []
+    # Check clusters
+    if all_clusters:
+        for cluster in all_clusters:
+            user_hash = cluster.get('user_hash')
+            if user_hash and user_hash not in user_ids:
+                missed_users.add(user_hash)
+                active_cluster_names.append(cluster['name'])
+        if active_cluster_names:
+            cluster_list = ', '.join(active_cluster_names)
+            resource_errors.append(
+                f'{len(active_cluster_names)} active cluster(s):'
+                f' {cluster_list}')
+
+    # Check managed jobs
+    if all_managed_jobs:
+        for job in all_managed_jobs:
+            user_hash = job.get('user_hash')
+            if user_hash and user_hash not in user_ids:
+                missed_users.add(user_hash)
+                active_job_names.append(str(job['job_id']))
+        if active_job_names:
+            job_list = ', '.join(active_job_names)
+            resource_errors.append(f'{len(active_job_names)} active'
+                                   f' managed job(s): {job_list}')
+
+    resource_error_summary = ''
+    if resource_errors:
+        resource_error_summary = ' and '.join(resource_errors)
+    missed_users_names = []
+    if missed_users:
+        all_users = global_user_state.get_all_users()
+        missed_users_names = [
+            user.name if user.name else user.id
+            for user in all_users
+            if user.id in missed_users
+        ]
+    return resource_error_summary, missed_users_names
+
+
+def _get_active_resources_for_workspaces(
+    workspace_names: List[str]
+) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    """Get active clusters or managed jobs for workspaces.
+
+    Args:
+        workspace_names: List of workspace_name.
+
+    Returns:
+        all_clusters: List[Dict[str, Any]]
+        all_managed_jobs: List[Dict[str, Any]]
+    """
+    if not workspace_names:
+        return [], []
+
+    def filter_by_workspaces(workspace_names: List[str]):
+        return lambda resource: (resource.get(
+            'workspace', constants.SKYPILOT_DEFAULT_WORKSPACE) in
+                                 workspace_names)
+
+    return _get_active_resources_by_names(workspace_names, filter_by_workspaces)
+
+
+def _get_active_resources_by_names(
```

Comment:
The function name sounds misleading, looks we're actually doing `get_active_resources_by_workspaces`